src/lib/services/openapi-service.ts

```diff
@@ -69,13 +69,18 @@ export class OpenApiService {
         res: Response<T>,
         schema: SchemaId,
         data: T,
+        headers: { [header: string]: string } = {},
     ): void {
         const errors = validateSchema(schema, data);
 
         if (errors) {
             this.logger.debug('Invalid response:', errors);
         }
 
+        Object.entries(headers).forEach(([header, value]) =>
+            res.header(header, value),
+        );
+
```

Comment:
Allow headers in `respondWithValidation` to open for correct 201 responses. The optional parameter means nothing breaks, but it might be worth spinning this out into a separate function or something?

Reply:
I won't complain if you do break this out but if you hadn't mentioned this in a comment it would't have worried me

Reply:
Got it! I'll ... I'll see what I feel like 😄 I think it does make sense that you can pass headers regardless of the status code, so maybe it makes sense to keep it?